File: C++/1052. Grumpy Bookstore Owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& customers, vector<int>& grumpy, int minutes) {
        int n=customers.size();
        int currentUnsatifiedCustomer=0;
        for(int i=0;i<minutes;i++){
            currentUnsatifiedCustomer+=customers[i]*grumpy[i];
        }
        int maximumUnsatifiedCustomer=currentUnsatifiedCustomer;
        int i=0;
        int j=minutes;
        while(j<n){
            currentUnsatifiedCustomer+=customers[j]*grumpy[j];
            currentUnsatifiedCustomer-=customers[i]*grumpy[i];            maximumUnsatifiedCustomer=max(maximumUnsatifiedCustomer,currentUnsatifiedCustomer);
            i++;
            j++;
        }
        int totalUnsatisfiedCust=maximumUnsatifiedCustomer;
        for(int i=0;i<n;i++){
            if(grumpy[i]==0){
                totalUnsatisfiedCust+=customers[i];
            }
        }
        return totalUnsatisfiedCust;
    }
};

File: JAVA/1052. Grumpy Bookstore Owner.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int n=customers.length;
        int cur=0,max=0;
        for(int i=0;i<minutes;i++){
            cur+=customers[i]*grumpy[i];
        }
        max=cur;
        int i=0,j=minutes;
        while(j<n){
            cur+=customers[j]*grumpy[j];
            cur-=customers[i]*grumpy[i];
            max=Math.max(cur,max);
            i++;
            j++;
        }
        int totalSatisfied=max;
        for(int z=0;z<n;z++){
            
           totalSatisfied+=customers[z]*(1-grumpy[z]); 
        }
        return totalSatisfied;
    }
}
